Xamarin.Forms.Platform.Android/AppCompat/FormsFragmentPagerAdapter.cs

```diff
@@ -31,7 +42,7 @@ public override long GetItemId(int position)
 		public override int GetItemPosition(Object objectValue)
 		{
 			var fragContainer = objectValue as FragmentContainer;
-			if (fragContainer != null && fragContainer.Page != null)
+			if (fragContainer?.Page != null)
```

Comment:
I think inside this code we need to remove the fragContainer from the _fragments List if it's not found. Otherwise when you remove a tab the _fragments list will still retain a references to fragContainer. 

```C#

		public override int GetItemPosition(Object objectValue)
		{
			var fragContainer = objectValue as FragmentContainer;
			if (fragContainer?.Page != null)
			{
				int index = _page.Children.IndexOf(fragContainer.Page);
				if (index >= 0)
					return index;
			}
            _fragments.Remove(fragContainer);
			return PositionNone;
		}
```

Reply:
I just tested this modification.
It seem to pass tests 4973 and 1675 with success, so I think it's ok to do that.
But in this case the fragment is not removed by the FragmentManager in the dispose, don't know if it's a good thing or not. 
At least it doesn't cause crash like now.

Reply:
From my tests when adding and removing pages the fragments related to those pages get removed from the FragmentManager.